ompi/mpi/c/pack_external.c.in

```diff
@@ -0,0 +1,74 @@
+/* -*- Mode: C; c-basic-offset:4 ; indent-tabs-mode:nil -*- */
+/*
+ * Copyright (c) 2004-2007 The Trustees of Indiana University and Indiana
+ *                         University Research and Technology
+ *                         Corporation.  All rights reserved.
+ * Copyright (c) 2004-2020 The University of Tennessee and The University
+ *                         of Tennessee Research Foundation.  All rights
+ *                         reserved.
+ * Copyright (c) 2004-2008 High Performance Computing Center Stuttgart,
+ *                         University of Stuttgart.  All rights reserved.
+ * Copyright (c) 2004-2005 The Regents of the University of California.
+ *                         All rights reserved.
+ * Copyright (c) 2006      Cisco Systems, Inc.  All rights reserved.
+ * Copyright (c) 2013      Los Alamos National Security, LLC.  All rights
+ *                         reserved.
+ * Copyright (c) 2015-2017 Research Organization for Information Science
+ *                         and Technology (RIST). All rights reserved.
+ * Copyright (c) 2024      Triad National Security, LLC. All rights
+ *                         reserved.
+ * $COPYRIGHT$
+ *
+ * Additional copyrights may follow
+ *
+ * $HEADER$
+ */
+#include "ompi_config.h"
+#include <stdio.h>
+
+#include "ompi/mpi/c/bindings.h"
+#include "ompi/runtime/params.h"
+#include "ompi/communicator/communicator.h"
+#include "ompi/errhandler/errhandler.h"
+#include "ompi/datatype/ompi_datatype.h"
+#include "opal/datatype/opal_convertor.h"
+#include "ompi/memchecker.h"
+
+PROTOTYPE ERROR_CLASS pack_external(STRING datarep, BUFFER inbuf, COUNT incount,
+                                    DATATYPE datatype, BUFFER_OUT outbuf,
+                                    AINT_COUNT outsize, AINT_COUNT_OUT position)
+{
+    int rc = MPI_SUCCESS;
+
+    MEMCHECKER(
+        memchecker_datatype(datatype);
+        memchecker_call(&opal_memchecker_base_isdefined, inbuf, incount, datatype);
+    );
+
+    if (MPI_PARAM_CHECK) {
+        OMPI_ERR_INIT_FINALIZE(FUNC_NAME);
+        if ((NULL == outbuf) || (NULL == position)) {  /* inbuf can be MPI_BOTTOM */
+            return OMPI_ERRHANDLER_NOHANDLE_INVOKE(MPI_ERR_ARG, FUNC_NAME);
+        } else if (incount < 0) {
+            return OMPI_ERRHANDLER_NOHANDLE_INVOKE(MPI_ERR_COUNT, FUNC_NAME);
+        } else if (outsize < 0) {
+            return OMPI_ERRHANDLER_NOHANDLE_INVOKE(MPI_ERR_ARG, FUNC_NAME);
+        }
+        OMPI_CHECK_DATATYPE_FOR_SEND(rc, datatype, incount);
+        OMPI_ERRHANDLER_NOHANDLE_CHECK(rc, rc, FUNC_NAME);
+        OMPI_CHECK_USER_BUFFER(rc, inbuf, datatype, incount);
+        OMPI_ERRHANDLER_NOHANDLE_CHECK(rc, rc, FUNC_NAME);
+#if OMPI_BIGCOUNT_SRC
+        OMPI_CHECK_MPI_COUNT_INT_CONVERSION_OVERFLOW(rc, incount);
```

Comment:
I don't quite grok what this is doing.  It's checking if `incount` -- which will be `MPI_Count` because of the `#if OMPI_BIGCOUNT_SRC` protection -- is larger than `INT_MAX`.  If it is, error.

Is `incount` not allowed to be greater than `INT_MAX`, even though it's `MPI_Count`?  If so, what is the value of it being an `MPI_Count`?

Reply:
That's right.  Until we get #13152 addressed we should have this protection I think.

Reply:
Given that #13152 was resolved by merging #13158, do these protections need to be removed now?

Reply:
those checks have been removed now.